gradle-plugin/src/main/java/com/squareup/anvil/plugin/AnvilExtension.kt

```diff
@@ -93,6 +93,26 @@ public abstract class AnvilExtension @Inject constructor(
   public val addOptionalAnnotations: Property<Boolean> = objects.property(Boolean::class.java)
     .conventionFromProperty("com.squareup.anvil.addOptionalAnnotations", false)
 
+  /**
+   * Enables incremental compilation support.
+   *
+   * This is achieved by maintaining a link between source files that exist before Anvil is run
+   * and the Anvil-generated files. This allows for two new behaviors:
+   * - Generated code is "invalidated" and deleted when the source file is changed or deleted.
+   * - Generated code is cached in a way that Gradle understands,
+   *   and will be restored from cache along with other build artifacts.
+   *
+   * By default this feature is enabled.
```

Comment:
I'm thinking maybe we should default this to disabled first and have it be opt-in while we get feedback on it, verify no major issues pop up, and also do benchmarking for performance impact. Then we can switch it to enabled by default in a follow-up release

Reply:
That's my thought as well, except that while testing against a snapshot or otherwise-special build, it's more convenient to just have it enabled.  So I was thinking leave it enabled by default up until release, then disable it by default, then enable it in a follow-up.

Reply:
> That's my thought as well, except that while testing against a snapshot or otherwise-special build, it's more convenient to just have it enabled. So I was thinking leave it enabled by default up until release, then disable it by default, then enable it in a follow-up.

It would be easy to get released in the wrong state that way though, and just creates additional work along the way. Shouldn't enabling for the snapshot be really easy now by just setting it in your gradle.properties file?

Reply:
Adding on to the previous comment, I think changing to false now is also preferred so that we're testing a snapshot as close to release as possible which lets us verify the expected default behavior etc